Cython/Compiler/Code.py

```diff
@@ -1514,20 +1514,17 @@ def generate_string_constants(self):
         if py_strings:
             self.use_utility_code(UtilityCode.load_cached("InitStrings", "StringTools.c"))
             py_strings.sort()
+
+            decls_writer.putln("/* String Constant Indexes */")
+            decls_writer.putln("enum {")
+            for _, _, py_string in py_strings:
+                decls_writer.putln("%s," % py_string.cname)
+            decls_writer.putln("%s" % Naming.string_tab_length_cname)
+            decls_writer.putln("};")
+
             w = self.parts['pystring_table']
             w.putln("")
-            w.putln("static __Pyx_StringTabEntry %s[] = {" % Naming.stringtab_cname)
-            w.putln("#if CYTHON_USE_MODULE_STATE")
-            w_in_module_state = w.insertion_point()
-            w.putln("#else")
-            w_not_in_module_state = w.insertion_point()
-            w.putln("#endif")
-            decls_writer.putln("#if !CYTHON_USE_MODULE_STATE")
-            not_limited_api_decls_writer = decls_writer.insertion_point()
-            decls_writer.putln("#endif")
-            init_globals.putln("#if CYTHON_USE_MODULE_STATE")
-            init_globals_in_module_state = init_globals.insertion_point()
-            init_globals.putln("#endif")
+            w.putln("static const __Pyx_StringTabEntry %s[] = {" % Naming.stringtab_cname)
```

Comment:
Ah, nice – this actually allows us to keep the string table as `static const`. That is a good thing.